libsolidity/lsp/LanguageServer.cpp

```diff
@@ -266,10 +311,31 @@ void LanguageServer::handleInitialize(MessageID _id, Json::Value const& _args)
 	replyArgs["capabilities"]["implementationProvider"] = true;
 	replyArgs["capabilities"]["textDocumentSync"]["change"] = 2; // 0=none, 1=full, 2=incremental
 	replyArgs["capabilities"]["textDocumentSync"]["openClose"] = true;
+	replyArgs["capabilities"]["semanticTokensProvider"]["legend"] = semanticTokensLegend();
+	replyArgs["capabilities"]["semanticTokensProvider"]["range"] = false;
+	replyArgs["capabilities"]["semanticTokensProvider"]["full"] = true; // XOR requests.full.delta = true
 
 	m_client.reply(_id, move(replyArgs));
 }
 
+void LanguageServer::semanticTokensFull(MessageID _id, Json::Value const& _args)
+{
+	auto uri = _args["textDocument"]["uri"];
+
+	compile();
+
+	auto const sourceName = m_fileRepository.uriToSourceUnitName(uri.as<string>());
+	SourceUnit const& ast = m_compilerStack.ast(sourceName);
+	m_compilerStack.charStream(sourceName);
+	SemanticTokensBuilder semanticTokensBuilder;
+	Json::Value data = semanticTokensBuilder.build(ast, m_compilerStack.charStream(sourceName));
```

Comment:
```suggestion
	Json::Value data = SemanticTokensBuilder().build(ast, m_compilerStack.charStream(sourceName));
```
Shouldn't that do it, too?